lib/ansible/plugins/inventory/docker_swarm.py

```diff
@@ -0,0 +1,168 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2018, Stefan Heitmueller <stefan.heitmueller@gmx.com>
+# Copyright (c) 2018 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    name: docker_swarm
+    plugin_type: inventory
+    authors:
+      - Stefan Heitmüller (stefan.heitmueller@gmx.com)
+    short_description: Ansible dynamic inventory plugin for Docker swarm nodes.
+    requirements:
+        - python >= 2.7
+        - Docker SDK for Python > 1.10.0
+    description:
+        - Reads inventories from the Docker swarm API.
+        - Uses a YAML configuration file docker_swarm.[yml|yaml].
+    options:
+        plugin:
+            description: The name of this plugin, it should always be set to 'docker_swarm' for this plugin to recognize it as it's own.
+            type: str
+            required: true
+            choices: docker_swarm
+        host:
+            description: Socket of a Docker swarm manager node (tcp,unix).
+            type: str
+            required: true
+        verbose_output:
+            description: Toggle to (not) include all available nodes metadata (e.g. Platform, Architecture, OS, EngineVersion)
+            type: bool
+            default: yes
```

Comment:
In my opinion, it should be set to `no` by default, and returns extended info only by user request. @felixfontein might disagree but at least that's what we do in docker modules

Reply:
Honestly, i don't mind either one or the other as default as long as it's configurable ;)

Reply:
There are only two docker modules which do that: docker_swarm_facts and docker_node_facts. And there it's only for lists of objects returned. This is very different to for example the output of docker_node_facts, and all other _facts modules. So I don't think this is a good reason why the default should be `no`. @morph027 since we can't make up our minds, just pick whatever *you* prefer.

Reply:
I'd like to say `yes` cause the data already is present, so no extra queries involved. And i don't think the amount of data is going to blow up memory ;)

Reply:
That's why I would also say `yes`. If someone thinks this is too much data, that person can simply turn it off with the option.